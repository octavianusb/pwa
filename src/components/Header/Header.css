.header {
  background: linear-gradient(to right, #24A3CF, #3650A0);
}

.container {
  display: flex;
  justify-content: space-between;
  max-width: var(--site-width);
  margin: 0 auto;
  padding: 0 15px;
}

.title {
  color: white;
  text-transform: uppercase;
  font-style: italic;
  padding: 0;
  font-size: 3.5rem;
}

.list {
  height: 100%;
  display: flex;
  align-items: center;
}

.link {
  color: white;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 2rem;
  margin: 0 1rem;
}

.link:hover, .link:focus {
  color: #2 4A3CF;
}