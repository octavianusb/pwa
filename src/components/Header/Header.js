import React from 'react'
import { NavLink } from 'react-router-dom'

// CSS
import styles from './Header.css'

export default function Header () {
  return (
    <header className={styles.header}>
      <div className={styles.container}>
        <h1 className={styles.title}>{'PWA'}</h1>
        <nav>
          <ul className={styles.list}>
            <li className={styles.listItem}><NavLink activeClassName='selected' className={styles.link} to='/'>{'Posts'}</NavLink></li>
            <li className={styles.listItem}><NavLink activeClassName='selected' className={styles.link} to='/about'>{'About'}</NavLink></li>
          </ul>
        </nav>
      </div>
    </header>
  )
}
