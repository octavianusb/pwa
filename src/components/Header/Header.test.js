import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Header from './Header'

describe('Header component', () => {
  it('should render as expected', () => {
    const component = shallow(<Header />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})