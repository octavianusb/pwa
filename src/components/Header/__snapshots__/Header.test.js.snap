// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render as expected 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <h1
      className="title"
    >
      PWA
    </h1>
    <nav>
      <ul
        className="list"
      >
        <li
          className="listItem"
        >
          <NavLink
            activeClassName="selected"
            className="link"
            to="/"
          >
            Posts
          </NavLink>
        </li>
        <li
          className="listItem"
        >
          <NavLink
            activeClassName="selected"
            className="link"
            to="/about"
          >
            About
          </NavLink>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
